models/yang/sonic/sonic-acl.yang

```diff
@@ -0,0 +1,221 @@
+module sonic-acl {
+	namespace "http://github.com/Azure/sonic-acl";
+	prefix sacl;
+	yang-version 1.1;
+
+	import ietf-inet-types {
+		prefix inet;
+	}
+
+	import sonic-common {
+		prefix cmn;
+	}
+
+	import sonic-port {
+		prefix prt;
+	}
+
+	import sonic-mirror-session {
+		prefix sms;
+	}
+
+	organization
+		"SONiC";
+
+	contact
+		"SONiC";
+
+	description
+		"SONIC ACL";
+
+	revision 2019-05-15 {
+		description
+			"Initial revision.";
+	}
+
+	container sonic-acl {
+
+		container ACL_TABLE {
+
+			list ACL_TABLE_LIST {
+				key "aclname";
+
+				leaf aclname {
+					type string {
+						pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,71})';
```

Comment:
The backend only consumes a specific set of ACL names. 
Q:Do we identify and restrict ?
Con: Polluting the DB. User would have tough time debugging, why his change is not taking effect, if he has mis-spelled it.
e.g.
1) "ACL_TABLE|SSH_ONLY"
2) "ACL_TABLE|DATAACL"
3) "ACL_TABLE|EVERFLOWV6"
4) "ACL_TABLE|SNMP_ACL"
5) "ACL_TABLE|EVERFLOW"

Reply:
It is inappropriate for ACLOrch to convert all keys/attribute/values to upper case before use.

However, if it is the current behavior, why we need to store ACL config in Redis DB in mixed case ?  Everything should be in upper case including  attributes name in schema and constraints need to be placed for all values (string/enum) to take only in upper case.

Reply:
> The backend only consumes a specific set of ACL names.
> Q:Do we identify and restrict ?
> Con: Polluting the DB. User would have tough time debugging, why his change is not taking effect, if he has mis-spelled it.
> e.g.
> 
> 1. "ACL_TABLE|SSH_ONLY"
> 2. "ACL_TABLE|DATAACL"
> 3. "ACL_TABLE|EVERFLOWV6"
> 4. "ACL_TABLE|SNMP_ACL"
> 5. "ACL_TABLE|EVERFLOW"

For specific ACL tables names patterns needs to be defined or we can define enum if it is always set of ACL tables names.

Reply:
As per internal discussion, we believe that the lower/upper case is a separate but common problem for many features. Implementing in one or other way, can impact multiple applications or UI. This would need separate discussion and we have less time to work on the current review.

Reply:
As per ACL SME, acl table name are never converted to uppercase, so it is case sensitive.  EVERRFLOWV6/everflowv6/EverflowV6/EverFlowV will have different entry in ACLOrch also.

Reply:
Regarding specific ACL names like ACL_TABLE|SSH_ONLY, there is no restriction, it is as per user config. So this comment is not applicable.

Reply:
**Not Applicable as per design.**